sys/include/net/ndp.h

```diff
@@ -344,9 +344,19 @@ typedef struct __attribute__((packed)) {
     uint8_t len;            /**< length in units of 8 octets */
     network_uint16_t resv;  /**< reserved field */
     network_uint32_t ltime; /**< lifetime in seconds */
-    ipv6_addr_t addrs[];    /**< addresses of IPv6 recursive DNS servers */
 } ndp_opt_rdnss_t;
 
+/**
+ * @brief   Get the start of addresses array in the recursive DNS server option
+ *
+ * @param[in]   opt   Recursive DNS server option in contiguous memory
+ *
+ * @returns     pointer to first address
+ */
+static inline ipv6_addr_t *ndp_opt_rdnss_addrs_ptr(ndp_opt_rdnss_t *opt) {
+	return (ipv6_addr_t *) (((uint8_t *)opt) + sizeof(ndp_opt_rdnss_t));
```

Comment:
Not MISRA-compliant, but I prefer the 

```suggestion
	return (ipv6_addr_t *) (opt + 1);
```

As it is way more readable in my opinion (but I know pointer arithmetics, so I can accept that people who don't might find this confusing).